controller/api/destination/endpoint_translator.go

```diff
@@ -330,11 +330,13 @@ func toWeightedAddr(address watcher.Address, opaquePorts, skippedInboundPorts ma
 	// hinted that this destination knows H2 (and handles our orig-proto
 	// translation)
 	var hint *pb.ProtocolHint
-	if enableH2Upgrade && controllerNSLabel != "" && !isSkippedInboundPort {
-		hint = &pb.ProtocolHint{
-			Protocol: &pb.ProtocolHint_H2_{
-				H2: &pb.ProtocolHint_H2{},
-			},
+	if controllerNSLabel != "" && !isSkippedInboundPort {
+		if enableH2Upgrade {
+			hint = &pb.ProtocolHint{
```

Comment:
`hint` is only initialized in this conditional, but it's used below in a separation condition, which may segfault.

We need to initialize the `hint` outside of the `enableH2Upgrade`...